core/src/definitions.ts

```diff
@@ -72,7 +72,37 @@ export interface PluginConfig {
   name: string;
 }
 
+export enum ExceptionCode {
+  /**
+   * API is not implemented.
+   *
+   * This means the API can't be used because:
+   *   - it is not implemented at all
+   *   - it is not implemented for the current platform
+   */
+  Unimplemented = 'UNIMPLEMENTED',
+
+  /**
+   * API is not available.
+   *
+   * This means the API can't be used right now because:
+   *   - it is currently missing a prerequisite, such as network connectivity
+   *   - it requires a particular platform or browser version
+   */
+  Unavailable = 'UNAVAILABLE',
+}
+
+export interface CapacitorException extends Error {
+  code: ExceptionCode;
+}
+
+declare const CapacitorException: {
+  prototype: CapacitorException;
+  new (message: string, code: ExceptionCode): CapacitorException;
```

Comment:
```suggestion
  new (message: string, code?: ExceptionCode): CapacitorException;
```
If we plan to make all errors thrown by Capacitor to be Capacitor.Exception, then code should be optional.
With current implementation and this changes, all native Capacitor errors thrown are Capacitor.Exception, but only unimplemented and unavailable have codes, other errors have undefined code. But on web we can't create Capacitor.Exception errors without a code